### PR TITLE
move input generation for copy benchmark to device

### PR DESCRIPTION
**Pull request by karthikeyann**
Use device functions to move input generation to device in copy benchmark.

Splitting PR https://github.com/rapidsai/cudf/pull/10109 for review

### CODE REVIEW

**Review by ttnghia — APPROVED** [review]

File: cpp/benchmarks/string/copy.cu [hunk 1]
```diff
@@ -24,7 +24,8 @@
 #include <cudf_test/column_wrapper.hpp>
 
 #include <algorithm>
-#include <random>
+
+#include <thrust/shuffle.h>
```
Comment (ttnghia) [inline cpp/benchmarks/string/copy.cu:28]: Nit: Move the `include` statement below to be above all other include statements.

Comment (karthikeyann) [reply cpp/benchmarks/string/copy.cu:27]: updated.

**Comment by karthikeyann** [comment]
rerun tests

File: cpp/benchmarks/string/copy.cu [hunk 2]
```diff
@@ -23,10 +24,9 @@
 #include <cudf/strings/strings_column_view.hpp>
 #include <cudf_test/column_wrapper.hpp>
 
-#include <algorithm>
-#include <random>
+#include <thrust/shuffle.h>
 
-#include "string_bench_args.hpp"
+#include <algorithm>
```
Comment (davidwendt) [inline cpp/benchmarks/string/copy.cu:29]:
```suggestion

```
I believe this was only needed for `std::iota` which has been removed here.

**Review by davidwendt — APPROVED** [review]

**Pull request merged**